### ISSUE TITLE
Context menu doesn't display properly inside modal

### ISSUE THREAD

**Issue by HoHieuLuc** [icflorescu/mantine-datatable#354]
**Describe the bug**
When using DataTable inside modal (@mantine/modals), row context menu displays at the the wrong position

**To Reproduce**

1. Go to https://codesandbox.io/s/naughty-haibt-ytxc86?file=/src/App.tsx
2. Click on open modal button
3. Right click on any row

**Expected behavior**
Context menu should display correctly

**Screenshots**

https://github.com/icflorescu/mantine-datatable/assets/62865718/36dd9063-2286-409d-896c-9ec643285dad

**Desktop (please complete the following information):**

- OS: Windows 10
- Browser: Brave
- Version v1.52.126

**Comment by icflorescu** [comment]
Good catch, thank you!
The fix will be available in a few minutes, as soon as `v2.7.1` hits the npm registry.
From now on, the context menu will be displayed inside a [`Portal`](https://mantine.dev/core/portal/) to make sure it shows properly when used on a table inside a [`Modal`](https://mantine.dev/core/modal/) component.

The only thing you'll have to do is to make sure you adjust its `zIndex` to a value higher that the modal's `zIndex`. Something greater than `201` should work, but I'd advise you to go higher just to be on the safe side:

```ts
<DataTable
  // data table props...
  rowContextMenu={{
    zIndex: 1000, // 👈 make sure the context menu is above the modal
    // context menu items...
  }}
/>
```

**Comment by icflorescu** [comment]
Done. Make sure to upgrade to `>2.7.1`.
I've also updated the docs, see [how to use row context menus inside modals](https://icflorescu.github.io/mantine-datatable/examples/row-context-menu#using-row-context-menus-inside-modals).
